src/lucky_router/fragment.cr

```diff
@@ -21,39 +21,37 @@
 # fragment.static_parts
 #
 # # Would return:
-# {"users" => Fragment, "posts" => Fragment}
+# {PathPart("users") => Fragment, PathPart("posts") => Fragment}
 #
 # # The Fragment in the 'users' key would have:
 #
-# # Fragment.new("id", dynamic: true)
+# # Fragment.new(PathPart(":id"))
 # fragment.dynamic_part
 #
 # # Static parts
 # fragment.static_parts
-# {"foo" => Fragment}
+# {PathPart("foo") => Fragment}
 # ```
 #
 # ## Gotcha
 #
 # The last fragment of a path is "empty". It does not have static parts or
 # dynamic parts
 class LuckyRouter::Fragment(T)
-  alias StaticPartName = String
   property dynamic_part : Fragment(T)?
-  getter static_parts = Hash(StaticPartName, Fragment(T)).new
+  getter static_parts = Hash(PathPart, Fragment(T)).new
   # Every path can have multiple request methods
   # and since each fragment represents a request path
   # the final step to finding the payload is to search for a matching request method
   getter method_to_payload = Hash(String, T).new
-  getter? dynamic
-  getter name : StaticPartName
+  getter path_part : PathPart
 
-  def initialize(@name, @dynamic = false)
+  def initialize(@path_part)
   end
 
   # This looks for a matching fragment for the given parts
   # and returns NoMatch if one is not found
-  def find(parts : Array(String), method : String) : Match(T) | NoMatch
+  def find(parts : Array(PathPart), method : String) : Match(T) | NoMatch
```

Comment:
I'm almost thinking that for `find` it can be what it was before with a String to maintain speed and may be more clear. To me the `PathPart` is mostly helpful when processing route definitions, but for route finding it seems like you end up just using the `name`.

What do you think about using just a `String` for finding? I could go either way but I think it might be clearer/faster